test/boost/index_with_paging_test.cc

```diff
@@ -18,11 +18,12 @@ SEASTAR_TEST_CASE(test_index_with_paging) {
         e.execute_cql("CREATE TABLE tab (pk int, ck text, v int, v2 int, v3 text, PRIMARY KEY (pk, ck))").get();
         e.execute_cql("CREATE INDEX ON tab (v)").get();
 
-        sstring big_string(4096, 'j');
+        sstring big_string(16384, 'j');
+        int nrows = 16 * 1024;
         // There should be enough rows to use multiple pages
         auto prepared_id = e.prepare("INSERT INTO tab (pk, ck, v, v2, v3) VALUES (?, ?, 1, ?, ?)").get0();
         auto big_string_v = cql3::raw_value::make_value(serialized(big_string));
-        max_concurrent_for_each(boost::irange(0, 64 * 1024), 50, [&] (auto i) {
+        max_concurrent_for_each(boost::irange(0, nrows), 50, [&] (auto i) {
```

Comment:
Sadly, I don't know if lowering the number of rows breaks the intention of this test, I'm afraid it might. This test was added in commit da3821c5980cc4da896159a1e9fc552d5e170d0b. The thing is, a SELECT using a secondary index has two steps for every page - one step is to get a page of **keys** from the view, and then to get a page of rows from **base table** based on these keys. Your change means that there are now only 16,000 keys to read from the row, so theoretically they may all fit in one 1MB page from the view (it doesn't help that the string **value** is bigger).

So I'm quite worried that after your patch, we might not be exercising the full index-scanning paging code.  By the way, Piotr Sarna specifically had a comment "// There should be enough rows to use multiple pages" about the number of rows, not their size.

Could you make the test faster not by reducing the number of rows, but by the size of each row? I don't think it's important that each "big_string" be 4K - I am guessing that even 100 bytes will make the base row significantly larger than the key so will exercise both types of paging in the scanning code.

Reply:
> I am guessing that even 100 bytes will make the base row significantly larger than the key so will exercise both types of paging in the scanning code.

This change alone brings the test from 9' to 6' in my local. Good enough.

Reply:
Done in https://github.com/scylladb/scylladb/commit/81a4178fceb5411447b6a8f021ca23f86fcfbb3a